lib/galaxy/datatypes/text.py

```diff
@@ -17,6 +17,62 @@
 log = logging.getLogger(__name__)
 
 
+class Biom(Text):
+    file_ext = "biom"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = get_file_peek(dataset.file_name,
+                                         is_multi_byte=is_multi_byte)
+            dataset.blurb = "Biological Observation Matrix"
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disc'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except:
+            return "BIOM file (%s)" % (nice_size(dataset.get_size()))
+
+
+class Biom1(Biom):
```

Comment:
Maybe it would make sense to derive from `Json` class instead (and drop `Biom` class definition). So `Biom1` will inherit `_looks_like_json()` method.